lib/rules/no-control-regex.js

```diff
@@ -83,8 +96,10 @@ module.exports = {
                 if (regex) {
                     const computedValue = regex.toString();
 
-                    if (hasControlCharacters(computedValue)) {
-                        context.report(node, "Unexpected control character in regular expression.");
+                    const controlCharacters = getControlCharacters(computedValue);
+
+                    if (controlCharacters.length > 0) {
+                        context.report(node, "Unexpected control character(s) in regular expression: " + controlCharacters.join(", ") + ".");
```

Comment:
@ilyavolodin If we do index or column, we need to report per character, not per literal. One of the test cases would need to have two errors. Not saying we can't do that, but it'd be great to make sure we agree on what @ljharb should implement here.

Personally I think displaying the literal text is already a huge win.